src/tools/pkcs15-init.c

```diff
@@ -3142,7 +3145,9 @@ static int verify_pin(struct sc_pkcs15_card *p15card, char *auth_id_str)
 			return SC_ERROR_OBJECT_NOT_FOUND;
 
 		if (pin_obj->label[0])
-			snprintf(pin_label, sizeof(pin_label), "User PIN [%s]", pin_obj->label);
+			snprintf(pin_label, sizeof(pin_label), "User PIN [%.*s]",
+				(int) MIN(sizeof(pin_label)-11, strlen(pin_obj->label)),
```

Comment:
same here.

Reply:
same as above.

I am not sure if this is the right way of doing that or even improvement. But makes it the text contained in the formatting string if we happen to have longer ones.

From my testing, it looks like I still need to add -1 to accommodate null byte, but otherwise it looks like working well:

```
$ cat test.c
#include <stdio.h>
#include <string.h>
#define MIN(x, y) (((x) < (y)) ? (x) : (y))
int main() {
	char label[255] = "some long label that we do not want to truncate";
	char pin_label[40];
	printf("sizeof(label[255]) = %d\n", sizeof(label));

	snprintf(pin_label, sizeof(pin_label), "User PIN [%s]", label); 
	printf("%s\n", pin_label);

	snprintf(pin_label, sizeof(pin_label), "User PIN [%.*s]",
		(int) MIN(sizeof(pin_label)-12, strlen(label)),
		label);
	printf("%s\n", pin_label);
}
$ gcc test.c
$ ./a.out 
sizeof(label[255]) = 255
User PIN [some long label that we do no
User PIN [some long label that we do n]
```